server/resources/resource_management.go

```diff
@@ -632,11 +656,26 @@ func DeleteResourceInTransaction(context middleware.Context, resourceSchema *sch
 	if !ok {
 		return fmt.Errorf("No environment for schema")
 	}
+
+	auth := context["auth"].(schema.Authorization)
+	policy := context["policy"].(*schema.Policy)
+	resource, err := mainTransaction.Fetch(resourceSchema, resourceID, policy.GetTenantIDFilter(schema.ActionDelete, auth.TenantID()))
+	if err != nil {
+		return err
+	}
+	if resource != nil {
+		context["resource"] = resource.Data()
+	}
+	// apply property filter
+	err = policy.ApplyPropertyConditionFilter(schema.ActionUpdate, resource.Data(), nil)
+	if err != nil {
+		return ResourceError{err, "", Unauthorized}
```

Comment:
Does anything need to happen to "resource" in case of error ? Should you use a "defer" statement after resource is assigned ? I don't have enough context to know whether the Fetch operation starts a transaction or just gets you a handle to a transaction that is in progress...